packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js

```diff
@@ -2580,6 +2580,34 @@ QUnit.module('Hidden input', {}, () => {
         assert.equal($hiddenInput.attr('name'), 'Editor with mask', 'name of hidden input');
     });
 
+    QUnit.test('Preserve value if the mask option is updated programmatically (T1214019)', function(assert) {
+        const $textEditor = $('#texteditor').dxTextEditor({
+            valueChangeEvent: 'input',
+            mask: '000',
+        });
+        const $input = $textEditor.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+        const instance = $textEditor.dxTextEditor('instance');
+
+        keyboardMock($input, true)
+            .caret(0)
+            .type('1')
+            .change();
+
+        $textEditor.blur();
+        instance.option('mask', '0000');
+        keyboardMock($input, true)
+            .caret(1)
+            .type('2')
+            .change();
+
+        $textEditor.blur();
+        instance.option('mask', '000');
+
+        const $hiddenInput = $textEditor.find('input[type=hidden]');
+
+        assert.equal($hiddenInput.val(), 12, 'Hidden value is saved');
```

Comment:
Why do you check a value of the hidden input here? I think we should check usual visible input's text.
I also prefer we check how `value` prop is changed because I guess it's unexpected too